### PR TITLE
Lazily load the LinuxDistribution data

### LINKED ISSUE #198: Considerable import-time side-effects
Importing distro performs (at the least) a subprocess call, this can be considerably slow and if distro isn't directly used could impact application startup time.

Most of this stems from the assignment of `distro._distro`:

https://github.com/nir0s/distro/blob/ff1699d63b3ba93411d339fa1b19836684c6f420/distro.py#L1077

This _could_ be avoided either by lazily instantiating the `_distro` object, or lazily assigning the attributes of `LinuxDistribution`.

I've attached a profile, here's some high level numbers though:

### status quo
```
$ time python -c 'import distro'

real	0m0.297s
user	0m0.112s
sys	0m0.024s
```

### removing module-time side-effects (naive)

```diff
--- a/distro.py
+++ b/distro.py
@@ -1074,7 +1074,7 @@ class LinuxDistribution(object):
         return distro_info
 
 
-_distro = LinuxDistribution()
+#_distro = LinuxDistribution()
 
 
 def main():
```

```
$ time python -c 'import distro'

real	0m0.019s
user	0m0.016s
sys	0m0.000s
```

### profile

Obtained using:
```
echo 'import distro' > test.py
python -m cProfile -o prof.pstats test.py
gprof2dot --skew=.1 -wn3 prof.pstats | dot -Tsvg -o prof.svg
```

Using [yelp's fork of gprof2dot (`pip install yelp-gprof2dot`)](https://github.com/Yelp/gprof2dot)

### PR DESCRIPTION
**Pull request by asottile**
Resolves #198

### CODE REVIEW

**Comment by nir0s** [comment]
Definitely better solution than mine.

File: distro.py [hunk 1]
```diff
@@ -511,6 +511,17 @@ def distro_release_attr(attribute):
     return _distro.distro_release_attr(attribute)
 
 
+class cached_property(object):
```
Comment (nir0s) [inline distro.py]: I think we should add docstrings here to explain what we're doing.

File: distro.py [hunk 2]
```diff
@@ -591,26 +602,20 @@ def __init__(self,
         self.os_release_file = os_release_file or \
             os.path.join(_UNIXCONFDIR, _OS_RELEASE_BASENAME)
         self.distro_release_file = distro_release_file or ''  # updated later
-        self._os_release_info = self._get_os_release_info()
-        self._lsb_release_info = self._get_lsb_release_info() \
-            if include_lsb else {}
-        self._distro_release_info = self._get_distro_release_info()
+        self.include_lsb = include_lsb
```
Comment (nir0s) [inline distro.py]: as it's a part of the API, it should be added to the documented.

File: distro.py (hunk 1 above)
Comment (asottile) [reply distro.py]: done!

File: distro.py (hunk 2 above)
Comment (asottile) [reply distro.py]: done!

**Pull request merged**